### ISSUE TITLE
TfTensorOutput casts floats as ints

### ISSUE THREAD

**Issue by alexnsfx** [bentoml/BentoML#813]
**Describe the bug**
`TfTensorOutput` casts floats as ints in tensor output because of `bentoml.adapters.utils.tf_tensor_2_serializable()`.

**To Reproduce**
1. Define a Tensorflow 2.0 model that returns float output
2. Define a BentoService that uses TfTensorOutput. TfTensorOutput is implicitly used by DefaultOutput when no output adapter is defined like below:
```python
@bentoml.env(pip_dependencies=["tensorflow==2.0", "numpy>=1.18.2", "pandas>=1.0.3"])
@bentoml.artifacts([TensorflowSavedModelArtifact("model")])
class MyModelService(bentoml.BentoService):

    @bentoml.api(input=TfTensorInput())
    def predict(self, inputs):
        return self.artifacts.model(inputs)
```
3. `pack()` your model with the service
4. Serve it with `bentoml serve MyModelService:latest`
5. Send a request
```bash
curl -X POST 'http://127.0.0.1:5000/predict' \
--header 'Content-Type: application/json' \
--data-raw '{"instances": [an-input-for-your-model]}'
```
6. Observe output: `[[0, 0, 0, 0]]`

**Expected behavior**
Output should be floats.
E.g. `[[-0.05740901082754135, 0.012774493545293808, 0.021846788004040718, -0.07579610496759415]]`

**Environment:**
 - OS: Ubuntu 18.04.4
 - Python 3.7.7
 - bentoml, version 0.8.1

**Additional context**

This is caused by the cast happening [here](https://github.com/bentoml/BentoML/blob/a134e3a/bentoml/adapters/utils.py#L41-L48).

int-cast is tried first and will succeed on float data.

Relying on [numpy.ndarray.dtype](https://numpy.org/devdocs/reference/generated/numpy.ndarray.dtype.html) to know the cast destination is probably safer.

**Comment by parano** [comment]
cc @bojiang

**Comment by parano** [comment]
Addressed in #823, will be available in release 0.8.2
